scripts/split_sentencepiece_id_file.py

```diff
@@ -0,0 +1,70 @@
+import click
+import hashlib
+import sys
+
+
+class SpmFileSplitter:
+    """
+    Does the work of assigning input lines at random to one of two
+    destinations, ensuring that multiple occurances of the same input
+    line (after normalizing away any whitespace differences) go to the
+    same destination.
+    """
+
+    def __init__(self, fraction, seed, digest_size=8):
+        self.digest_size = digest_size  # bytes (so x8 for bits)
+        self.threshold = int(0.5 + fraction * pow(2, 8 * self.digest_size))
+        self.seed = str(seed).encode("utf-8")
+
+    def __call__(self, line):
+        normalized_line = " ".join(line.lstrip().rstrip().split())
+        seeded_encoded_line = self.seed + normalized_line.encode("utf-8")
+        hash = int(
+            hashlib.blake2b(
```

Comment:
Oh neat -- do you recommend using blake2b instead of the other places we're using mmh3 in this project? That would save us a dependency.

Reply:
As I mentioned in code review last week, I don't know of any big advantage of blake2b over mmh3 (other than eliminating the third-party dependency).  I'd say leave mmh3 in place where it is for now, but whenever we touch a file where it is used we should consider replacing it.